### DISCUSSION TITLE
TickTickSync is now available in the Community Plugin store.

### DISCUSSION THREAD

**Discussion by thesamim** [thesamim/TickTickSync#11] [Announcements]
Looking forward to the feedback!

**Comment by trainlady** [comment]
I am so excited for this. Thank you.  I maybe installed it and did the sync without enough thought. 

The not thinking part: I didn't think that I used descriptions. I was wrong. Every task that is emailed in puts the body of the email in the description field. That's on me for not recognizing that.

The big thing that I found is that I wasn't planning for is the handling of the "Won't Do" category in TickTick. Those tasks are reset to undone before syncing in Obsidian so I have a bunch of active tasks that I hadn't expected to be active. 

I'm going to create a Won't Do list and sort them there.  But that might be a thing to put in the caveats.

Thank you again for doing this. I'm going to continue to use it and will let you know how it goes.

**Comment by thesamim** [comment]
Thank you for the feedback.

> The big thing that I found is that I wasn't planning for was is the handling of the "Won't Do" category in TickTick. Those tasks are reset to undone before syncing in Obsidian so I have a bunch of active tasks that I hadn't expected to be active.

Thanks for bringing this up. That's not a function I used, so I didn't realize TickTick has a different status for "Won't do". Which is why the status got reset. I've opened an [issue](https://github.com/thesamim/TickTickSync/issues/15) to take care of this in a future release.

Re: Description Field:
Other than not clobbering it (work in progress: will be in the next release), would it be useful to bring the descriptions over to Obsidian? If it is: 

1. Any thoughts on how to present the descriptions? My current thinking is inline after a task, but that could make task lists very messy very quickly.
3. Any benefits to description synchronization being two way?

**Reply by trainlady** [reply]
Hi,

Thank you for your prompt response. 

The piece of me that wants Obsidian to capture things that I think are important says I would love the descriptions in Obsidian. It would also provide context if I was working from Obsidian. However, I agree that it could make the task list awkward. 

I'm not sure how much I would look at a list in Obsidian anyway. The way I currently use tasks is to bring the few tasks that I intend to work on onto my daily page. I look at the list if I am adding to it. I imagine that I will do my scheduling in TickTick and work off my daily page in Obsidian. I won't know until I try.

I don't think that I would want the synchronization to be two way. I would want the notes in Obsidian. I won't be looking for them in TickTick. But that's the way that I work.

**Reply by thesamim** [reply]
Thanks for the feedback.

**Comment by vaburov** [comment]
Hi thesamim,
 
this is really interesting start, the most promising trial among other ticktick plugins!
However, I lost all my descriptions in the tasks which is painful. I know, that's my fault, I had to read carefully all the description, but but foolproofing is very important. Please, please, protect user data from any updates by default... Don't touch descriptions and don't add tags by default...  Ideally, i believe, by default it should do the minimum: only sync tasks in one direction from TickTick to Obsidian (only non-completed, I don't want to get all the tasks from the past!) and then sync completed/not-completed states in both directions. That would be quite universal, very easy to validate and use. The rest of the features I believe should be optional, partly experimental and definitely non-default.

Another request which should be very easy to implement: please, let the user to setup a folder in Obsidian Vault where to put ticktick lists/folders, I don't want to see them in the root. Ideally, I would like to be able to map each ticktick list/folder to a dedicated folder in Obsidian, if possible, but just setting up a new root folder would be very useful.

Thank you for doing this, at the moment I am not able to use it with my flows, but I am going to monitor the changes.

**Comment by thesamim** [comment]
Thank you!

> [...] However, I lost all my descriptions in the tasks which is painful. 

So sorry!

This will be fixed in the next release. Please track with [this issue](https://github.com/thesamim/TickTickSync/issues/10)

> [...] and don't add tags by default... 

Please expand on this. Why wouldn't tag tracking be useful?


> [...] (only non-completed, I don't want to get all the tasks from the past!) 

There is a [request](https://github.com/thesamim/TickTickSync/issues/7) to only sync tasks with a certain tag from TickTick. I **think** I should be able to also add the ability to only sync open tasks from TickTick.

> Another request which should be very easy to implement: please, let the user to setup a folder in Obsidian Vault where to put ticktick lists/folders, I don't want to see them in the root. Ideally, I would like to be able to map each ticktick list/folder to a dedicated folder in Obsidian, if possible, but just setting up a new root folder would be very useful.

Please see [this issue](https://github.com/thesamim/TickTickSync/issues/12) to track this request. 

 
> Thank you for doing this, at the moment I am not able to use it with my flows, but I am going to monitor the changes.

Thank you!

**Reply by vaburov** [reply]
> Please expand on this. Why wouldn't tag tracking be useful?

Sorry, I didn't mean that. Tag tracking might be useful (especially if there are options to choose which tags or to sync all of the tasks).  So, as I understand the scenario, I add tags for certain tasks and then they are synced. 

But what is happened (when I enabled all vault sync), is that all my tasks in TickTick were automatically tagged by plugin. Which was, I guess, not even mentioned in the option description.  Actually, I would like my TickTick tasks never been touched by plugin (by default), except changing the completion state.
